test/unit/buffer_spec.lua

```diff
@@ -293,11 +298,131 @@ describe('buffer functions', function()
+
+      local expected_width = 20
+      local width = build_stl_str_hl("neo%=vim", expected_width)
+
+      eq(width, expected_width)
+      eq("neo              vim", get_str(output_buffer, expected_width))
+    end)
+
+    it('should approximately center text when using %=text%=', function()
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local expected_width = 20
+      local width = build_stl_str_hl("abc%=neovim%=def", expected_width)
+
+      eq(width, expected_width)
+      eq("abc    neovim    def", get_str(output_buffer, expected_width))
+    end)
+
+    it('should completely fill the buffer when using %=text%=', function()
+      -- This tests to make sure that it still fills the buffer, even when
+      --  we don't have perfectly equal spacing
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local expected_width = 20
+      local width = build_stl_str_hl("abc%=neo_vim%=def", expected_width)
+
+      eq(width, expected_width)
+      eq("abc   neo_vim    def", get_str(output_buffer, expected_width))
+    end)
+
+    it('should correctly quadrant the text when using 3 %=\'s', function()
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local expected_width = 40
+      local width = build_stl_str_hl("abc%=n%=eovim%=def", expected_width)
+
+      eq(width, width)
+      eq("abc         n         eovim          def", get_str(output_buffer, expected_width))
+    end)
+
+    it('should truncate when standard text pattern is too long', function()
+      -- This is the standard way Vim truncates text when it is too long
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local pat = "abcneovimdef"
+      local expected_width = 10
+      local width = build_stl_str_hl(pat, expected_width)
+
+      eq(width, width)
+      eq("<neovimdef", get_str(output_buffer, expected_width))
+    end)
+
+    it('should truncate when using =', function()
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local pat = "abcdef%=ghijkl"
+      local expected_width = 10
+      local width = build_stl_str_hl(pat, expected_width)
+
+      eq(width, expected_width)
+      -- It puts all of the text that is left aligned in the buffer,
+      --    and then truncates the text on the right.
+      eq("abcdef<jkl",  get_str(output_buffer, expected_width))
+    end)
+
+    it('should truncate centered text when using %=%=', function()
+      -- Is this a desired behavior?
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local pat = "abc%=this_text_should_be_gone%=def"
+      local expected_width = 7
+      local width = build_stl_str_hl(pat, expected_width)
+
+      eq(width, expected_width)
+      eq("abc<def", get_str(output_buffer, expected_width))
+    end)
```

Comment:
@tjdevries And about “how well”: do you know about such plugins as powerline and airline? They use Unicode code points that are not supported by absolutely all existing single-byte encodings: because they are in private use area.